scapy/arch/unix.py

```diff
@@ -78,8 +94,22 @@ def read_routes():
         if not "G" in flg:
             gw = '0.0.0.0'
         if netif is not None:
-            ifaddr = scapy.arch.get_if_addr(netif)
-            routes.append((dest,netmask,gw,netif,ifaddr))
+            try:
+                ifaddr = scapy.arch.get_if_addr(netif)
+                routes.append((dest,netmask,gw,netif,ifaddr))
+            except OSError as exc:
+                if exc.message == 'Device not configured':
+                    # This means the interface name is probably truncated by
+                    # netstat -nr. We attempt to guess it's name and if not we
+                    # ignore it.
+                    netif = _guess_iface_name(netif)
```

Comment:
It might be interesting to assign the returned value to fixed_netif in order to display a correct warning. See @Osso comment.